### PR TITLE
Add various plotting scripts for plasma profiles and branching ratios

### PR DESCRIPTION
**Pull request by chris-ashe**
This pull request includes several new plotting scripts for generating various documentation plots using `bokeh` and `matplotlib`. The changes introduce new visualizations and interactive elements to enhance data representation.

### New Plotting Scripts:

* [`documentation/proc-pages/scripts/plotting_scripts/beam_energy_decay.py`](diffhunk://#diff-c584cc44e982d7f0c6c9d15a49d0edaf7ed75c361854bf67b88a80f423c581b4R1-R48): Added a script to plot a parabolic profile with interactive sliders for plasma center value and profile index.
* [`documentation/proc-pages/scripts/plotting_scripts/deuterium_branching_plot.py`](diffhunk://#diff-5bbcead06b76888c5710bd8d6584aae00d562050d3bcc6343920b46ea7d4e5a8R1-R29): Added a script to plot the deuterium-deuterium branching ratio using `matplotlib`.
* [`documentation/proc-pages/scripts/plotting_scripts/fast_alpha_pressure.py`](diffhunk://#diff-83578272e6aa46f17889b985a8ec0711fbc4e24d000cafb683635daafae4496eR1-R59): Added a script to plot current profiles with interactive sliders for plasma center value.
* [`documentation/proc-pages/scripts/plotting_scripts/plot_pulsed_current.py`](diffhunk://#diff-0a58669c343515c4a806966cea11d8578aef845a4958cc553b8405da4ad06f8dR1-R125): Added a script to plot pulsed current profiles over time using `matplotlib`.
* [`documentation/proc-pages/scripts/plotting_scripts/profile_hastie_current.py`](diffhunk://#diff-6526024f2a0a3d556beb6df064afcb254fb46e9bbd99d505915a92f4c7cb58dbR1-R62): Added a script to plot the PROCESS vs Hastie current profile with interactive sliders for plasma center value and profile index.
* [`documentation/proc-pages/scripts/plotting_scripts/profile_parabolic_gradiant_plot.py`](diffhunk://#diff-6c494179549189e255c2291f1510216fd588a965aff7611dd50ca49fc1af3a48R1-R42): Added a script to plot the parabolic gradient profile with interactive sliders for plasma center value and profile index.
* [`documentation/proc-pages/scripts/plotting_scripts/profile_parabolic_plot.py`](diffhunk://#diff-c584cc44e982d7f0c6c9d15a49d0edaf7ed75c361854bf67b88a80f423c581b4R1-R48): Added a script to plot a parabolic profile with interactive sliders for plasma center value and profile index.
* [`documentation/proc-pages/scripts/plotting_scripts/profile_pedestal_plot.py`](diffhunk://#diff-0b3fa3b6624989956bc25bd4270b0d587d2045c70b2b16af29476940965a81b8R1-R77): Added a script to plot the pedestal profile with multiple interactive sliders for various parameters.
* [`documentation/proc-pages/scripts/plotting_scripts/profile_peng_qbar.py`](diffhunk://#diff-2d8381b0b1afffafb341e7cecaffc690141896d02b106f6fd4ce9dfecad39b55R1-R68): Added a script to plot the average safety factor for PROCESS vs STAR code with interactive sliders for safety factors.## Description

## Checklist

I confirm that I have completed the following checks:

- [x] My changes follow the [PROCESS style guide](https://ukaea.github.io/PROCESS/development/standards/)
- [x] I have justified any large differences in the regression tests caused by this pull request in the comments.
- [x] I have added new tests where appropriate for the changes I have made.
- [x] If I have had to change any existing unit or integration tests, I have justified this change in the pull request comments.
- [x] If I have made documentation changes, I have checked they render correctly.
- [x] I have added documentation for my change, if appropriate.

### CODE REVIEW

File: documentation/proc-pages/scripts/plotting_scripts/beam_energy_decay.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+""",
+)
+
+n0.js_on_change("value", callback)
+alpha.js_on_change("value", callback)
+
+# Save the plot as HTML
+save(
+    row(plot, column(n0, alpha)),
+    filename="/PROCESS/profile_parabolic_plot.html",
```
Comment (timothy-nunn) [inline documentation/proc-pages/scripts/plotting_scripts/beam_energy_decay.py:47]:
```suggestion
    filename="profile_parabolic_plot.html",
```

File: documentation/proc-pages/scripts/plotting_scripts/fast_alpha_pressure.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from bokeh.layouts import column, row
+from bokeh.models import ColumnDataSource, CustomJS, Slider
+from bokeh.plotting import figure, output_file, save
+
+x = np.linspace(0, 10, 500)
+y1 = 0.29 * (x) ** 2(0.75 - 0.37)
+y2 = 0.26(0.75 - 0.65) ** 0.5 * (x) ** 2  # Initial data for the second line
```
Comment (timothy-nunn) [inline documentation/proc-pages/scripts/plotting_scripts/fast_alpha_pressure.py:8]:
```suggestion
y1 = 0.29 * (x) ** 2 * (0.75 - 0.37)
y2 = 0.26 * (0.75 - 0.65) ** 0.5 * (x) ** 2  # Initial data for the second line
```

**Review by timothy-nunn — CHANGES_REQUESTED** [review]
Why don't all of the scripts (except the matplotlib ones) use `output_file`?

File: documentation/proc-pages/scripts/plotting_scripts/plot_pulsed_current.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ax.annotate("Between Pulse", xy=(6.5, -3.4), ha="center", va="center")
+
+# Add axis labels
+ax.set_xlabel("Time", fontsize=12)
+ax.xaxis.set_label_coords(0.97, 0.5)
+ax.set_ylabel("Current", fontsize=12)
+
+# Remove axes ticks
+ax.set_xticklabels([])
+ax.set_yticklabels([])
```
[the remark this answers is not in the document]
Comment (chris-ashe) [reply documentation/proc-pages/scripts/plotting_scripts/plot_pulsed_current.py:111]: Yes

File: documentation/proc-pages/scripts/plotting_scripts/profile_parabolic_plot.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+plot = figure(
+    x_range=(0, 1),
+    y_range=(0, 10),
+    width=400,
+    height=400,
+    title="Parabolic Profile | L-mode",
+)
+plot.xaxis.axis_label = r"Normalized Radius, $$ \rho $$"
+plot.yaxis.axis_label = r"Density, $$n_e$$"
```
[the remark this answers is not in the document]
Comment (chris-ashe) [reply documentation/proc-pages/scripts/plotting_scripts/profile_parabolic_plot.py:21]: You need to run the script with at least Bokeh 2.8.0

**Review by timothy-nunn — CHANGES_REQUESTED** [review]
Pre-commit failure: https://github.com/ukaea/PROCESS/pull/3412/checks#step:5:39

This comment still applies: https://github.com/ukaea/PROCESS/pull/3412#discussion_r1869311849

File: documentation/proc-pages/scripts/plotting_scripts/profile_pedestal_plot.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+""",
+)
+
+
+T0.js_on_change("value", callback)
+alpha.js_on_change("value", callback)
+Trho.js_on_change("value", callback)
+Tped.js_on_change("value", callback)
+Tsep.js_on_change("value", callback)
+Tbeta.js_on_change("value", callback)
```
[the remark this answers is not in the document]
Comment (chris-ashe) [reply documentation/proc-pages/scripts/plotting_scripts/profile_pedestal_plot.py:70]: Fixed

**Review by timothy-nunn — APPROVED** [review]

**Pull request merged**